src/viz/RobotModel.js

```diff
@@ -100,16 +104,40 @@ class RobotModel extends URDFLoader {
   }
 
   destroy() {
+    const robotModelParent = this.robotModel && this.robotModel.parent;
+    if (robotModelParent) {
+      // TODO: proper scene clear for all viz types
+      // just parent.remove is not enough
+      Object.values(this.robotModel.links).map(linkObj => {
+        if (linkObj.parent) {
+          linkObj.parent.remove(linkObj);
```

Comment:
The link objects should not be removed. There should still be empty groups in the scene or the tf hierarchy will be destroyed which can affect other visualizations.
Ideally I guess the way to handle this is by having destroy functions in the urdf classes that remove properties and meshes specific to urdf while leaving empty groups behind